File: c++代码/19.删除链表的倒数第N个节点.cpp
#include <iostream>
#include <string>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution
{
public:
    ListNode *removeNthFromEnd(ListNode *head, int n)
    {
        ListNode *left_node = head;
        ListNode *right_node = head;
        for (int i = 0; i < n - 1; i++)
        {
            right_node = right_node->next;
        }
        if (!right_node->next)
        {
            return head->next;
        }

        ListNode *tmp_node = NULL;
        while (right_node->next)
        {
            tmp_node = left_node;
            left_node = left_node->next;
            right_node = right_node->next;
        }
        tmp_node->next = left_node->next;
        return head;
    }
};

int main()
{
    return 0;
}

File: 力扣/链表/19. 删除链表的倒数第N个节点.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        left_node = right_node = head
        for i in range(n):
            right_node = right_node.next
        if not right_node.next:
            return left_node.next
        while right_node.next:
            left_node = left_node.next
            right_node = right_node.next
